### PR TITLE
[chromium] Implement autocomplete's Login

### LINKED ISSUE #1179: [Chromium] Enable password save/autofill
There is no implementation in the Chromium backed for the login save/autofill

### PR DESCRIPTION
**Pull request by mshin-wolvic**
- Implement Login UI & Login Selection UI in PromptDelegateImpl
- Implement Login DB in AutocompleteDelegateWrapper/AutocompleteStorageProxy/ChromiumLoginStorageDelegate

Fixes #1179

### CODE REVIEW

**Review by zakharvoit — APPROVED** [review]

File: app/src/common/chromium/com/igalia/wolvic/browser/api/impl/PromptDelegateImpl.java [hunk 1]
```diff
@@ -823,4 +833,239 @@ public WSession.PromptDelegate.PromptResponse dismiss() {
...
+                } else {
+                    resetLoginSelectPrompt();
+                    return false;
+                }
+            } catch (WindowManager.BadTokenException e) {
+                resetLoginSelectPrompt();
+                return false;
+            }
+            return true;
+        }
```
Comment (svillar) [inline app/src/common/chromium/com/igalia/wolvic/browser/api/impl/PromptDelegateImpl.java:959]: This method is nearly identical to the above one, can't we factorize them in a private method? Maybe not the whole method but at least starting from the `mLoginSelectPrompt`  assignment it should be quite easy to refactor.

**Review by svillar — CHANGES_REQUESTED** [review]
Looks great, awesome work, I just have two minor comments

File: app/src/common/chromium/com/igalia/wolvic/browser/api/impl/AutocompleteStorageProxy.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        mDelegate.onLoginUsed(fromPasswordForm(form), WAutocomplete.UsedField.PASSWORD);
+    }
+
+    public WResult<Boolean> checkLoginIfAlreadySaved(PasswordForm form) {
+        final WResult<WAutocomplete.LoginEntry> found = mDelegate.findLoginByGuid(form.getGuid());
+        if (found == null) {
+            return WResult.fromValue(false);
+        }
+        return found.then(result -> {
+            return WResult.fromValue(result != null ? true : false);
```
[the remark this answers is not in the document]
Comment (mshin-wolvic) [reply app/src/common/chromium/com/igalia/wolvic/browser/api/impl/AutocompleteStorageProxy.java]: Done

File: app/src/common/chromium/com/igalia/wolvic/browser/api/impl/PromptDelegateImpl.java (hunk 1 above)
Comment (mshin-wolvic) [reply app/src/common/chromium/com/igalia/wolvic/browser/api/impl/PromptDelegateImpl.java:947]: Done

**Review by svillar — APPROVED** [review]
Great work. Merging now that the chromium side is in

**Pull request merged**